src/contracts/engine/AmguConsumer.ts

```diff
@@ -9,8 +9,7 @@ export class AmguConsumer extends Contract {
   }
 
   public async getAmguToken(block?: number) {
-    const result = await this.makeCall<Address>('mlnToken', undefined, block);
-    return result;
+    return await this.makeCall<Address>('mlnToken', undefined, block);
```

Comment:
No need for `return await` here. Just make the function NOT async and return the result from this.makeCall() directly. Otherwise we add an additional redundant promise wrapper.